### PR TITLE
[python] Add support for hot reload of python code in docker mode

### PR DESCRIPTION
**Pull request by eolivelli**
Summary:

When you are working locally, add the ability to automatically reload python code without restarting the container

In "docker run" mode the cli start watching on the python files.
In case of any python file that has changed the python agent is restarted, picking up the new version of the code.

The is a new flag "--watch-files=false" to disable the feature.

The watcher starts only if there is a "python" directory in the application

Follow up:
- if there is a syntax error the Python process crashes, so the docker container exists
- add test cases

### CODE REVIEW

File: langstream-runtime/langstream-runtime-impl/src/main/java/ai/langstream/runtime/agent/api/AgentInfoServlet.java [hunk 1]
```diff
@@ -20,19 +20,38 @@
...
     @Override
     protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException {
         resp.setContentType("application/json");
-        MAPPER.writeValue(resp.getOutputStream(), agentInfo.serveWorkerStatus());
+        MAPPER.writeValue(resp.getOutputStream(), agentAPIController.serveWorkerStatus());
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+        resp.setContentType("application/json");
```
Comment (eolivelli) [inline langstream-runtime/langstream-runtime-impl/src/main/java/ai/langstream/runtime/agent/api/AgentInfoServlet.java:43]: This is currently not used

File: langstream-cli/src/main/java/ai/langstream/cli/commands/docker/LocalRunApplicationCmd.java [hunk 2]
```diff
@@ -96,6 +100,11 @@ public class LocalRunApplicationCmd extends BaseDockerCmd {
             description = "Start the UI")
     private boolean startUI = true;
 
+    @CommandLine.Option(
+            names = {"--watch-files"},
+            description = "Start the UI")
```
Comment (nicoloboschi) [inline langstream-cli/src/main/java/ai/langstream/cli/commands/docker/LocalRunApplicationCmd.java:105]: to change

File: langstream-cli/src/main/java/ai/langstream/cli/commands/docker/LocalRunApplicationCmd.java [hunk 3]
```diff
@@ -380,26 +393,75 @@ private void executeOnDocker(
...
+                    }
+                },
+                watcher);
+    }
+
+    private static void restartAgents() {
+        HttpURLConnection urlConnection = null;
+        try {
+            URL url = new URL("http://localhost:8790/commands/restart");
+            System.out.println("Calling " + url + " to restart the agents");
```
Comment (nicoloboschi) [inline langstream-cli/src/main/java/ai/langstream/cli/commands/docker/LocalRunApplicationCmd.java:454]: can you use the cli logger?

Comment (eolivelli) [reply langstream-cli/src/main/java/ai/langstream/cli/commands/docker/LocalRunApplicationCmd.java]: sure

**Pull request merged**